hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -459,6 +459,46 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}
 
+	// Reconcile etcd client MTLS secret if the control plane is using an unmanaged etcd cluster
+	if hcluster.Spec.Etcd.ManagementType == hyperv1.Unmanaged {
+		r.Log.Info("Reconciling etcd client mtls secret")
+		if hcluster.Spec.Etcd.Unmanaged == nil || len(hcluster.Spec.Etcd.Unmanaged.TLS.Client.Name) == 0 || len(hcluster.Spec.Etcd.Unmanaged.Endpoint) == 0 {
+			return ctrl.Result{}, fmt.Errorf("etcd metadata not specified for unmanaged deployment")
+		}
+		var src corev1.Secret
+		if err := r.Client.Get(ctx, ctrlclient.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Etcd.Unmanaged.TLS.Client.Name}, &src); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to get etcd client cert %s: %w", hcluster.Spec.Etcd.Unmanaged.TLS.Client.Name, err)
+		}
+		if _, ok := src.Data["etcd-client.crt"]; !ok {
```

Comment:
My preference would be to report this somewhere in the HostedCluster status. Otherwise the API end user has no idea why their cluster is not progressing.

Reply:
Adjusted strategy to handle reporting